### ISSUE TITLE
Command_inout_asynch empty argout 

### ISSUE THREAD

**Issue by dmiego** [tango-controls/pytango#177]
I try to execute command_inout_asynch in PyTango client, but every time I've got an event with empty argout (and argout_raw). All other fields of the event are correct, e.g. event.device

Code example:
```
import PyTango
import time

class TestClass():

    def __init__(self):

        api_util = PyTango.ApiUtil.instance()
        # change client to "push" model
        api_util.set_asynch_cb_sub_model(PyTango.cb_sub_model.PUSH_CALLBACK) 
        self.dev = PyTango.DeviceProxy('sys/tg_test/1')

    def cmd(self):
        self.dev.command_inout_asynch('Status', self.callback)
        time.sleep(0.5)  # not immediate response, have to wait some time

    def callback(self, event):
        print(event)
        print(event.device)
        print(event.argout)
        print(event.argout_raw)

def main():

    test = TestClass()
    test.cmd()

if __name__ == '__main__':
    main()
```

Console output:
```
<tango._tango.CmdDoneEvent object at 0x04344CA8>
TangoTest(sys/tg_test/1)
None
DeviceData[
]
```

The problem replicates both on Windows and Linux, Python 2.7 and 3.4, Pytango 9.2.0 and 9.2.2.

P.S.: Also documentation about all kinds of "command_inout" dissapearred from http://pytango.readthedocs.io/en/latest/contents.html

**Comment by vxgmichel** [comment]
Thanks for your report,

I've been able to reproduce your issue, and I'm currently investigating it. Interestingly enough, getting the result using `proxy.command_inout_reply` does work.

```python
import time
from tango.test_context import DeviceTestContext
from tango.server import Device, command


class TestDevice(Device):

    @command(dtype_out=str)
    def say_hello(self):
        return "Hello world!"

with DeviceTestContext(TestDevice) as proxy:
    eid = proxy.command_inout_asynch('say_hello')
    time.sleep(0.1)
    print(proxy.command_inout_reply(eid))
```

I also fixed the docs for `DeviceProxy`, I'll include it in the upcoming PR.

**Comment by vxgmichel** [comment]
Also note that you can get the command result using:

```python
print(event.argout_raw.extract())
```

**Comment by vxgmichel** [comment]
Thanks for your report, the issue has been fixed and merged into the develop branch (#178).

It will become visible with pytango 9.2.3. You can use the `event.argout_raw.extract()` trick in the meantime.
